packages/jsii-java-runtime/project/src/main/java/software/amazon/jsii/JsiiEngine.java

```diff
@@ -198,6 +198,9 @@ public Object getObject(final JsonNode objRefNode) {
      * @return The Java class name.
      */
     private Class<?> resolveJavaClass(final String fqn) throws ClassNotFoundException {
+        if ("Object".equals(fqn)) {
```

Comment:
This might be mentioned separately in the commit/merge message. Seems like a different fix.